File: 프로그래머스/lv2/12914. 멀리 뛰기/멀리 뛰기.cs
using System;
public class Solution {
    long[] dp  = new long[2001];
    public long solution(int n) {
        long answer = 0;

        dp[0] = 0;
        dp[1] = 1;
        dp[2] = 2;
        
        for(int i=3;i<=n;++i)
        {
            dp[i] = (dp[i-1] + dp[i-2]) % 1234567;
            
        }
        answer = dp[n];
        
        return answer;
    }
}

File: 프로그래머스/2/12914. 멀리 뛰기/멀리 뛰기.cpp
#include <string>
#include <vector>
#include <iostream>
using namespace std;

long long sumList[2001];
void DFS(int sum, long long * result);
long long solution(int n) {
    long long answer = 0;
    
    sumList[1] = 1;
    sumList[2] = 2;
    for(int i=3;i<=n;++i)
    {
        sumList[i] = (sumList[i-2] + sumList[i-1])%1234567;
    }

    return sumList[n];
}